src/system/dev/vfs.c

```diff
@@ -155,6 +157,83 @@ int _close(int file) {
 	return ret;
 }
 
+int _mkdir(const char* path){
+    struct _reent* r = _REENT;
+    size_t i;
+
+    for(i = 0; i < MAX_FILELEN; i++) {
+        if (path[i] == '\0') break;
+    }
+
+    if(i >= MAX_FILELEN) {
+        r->_errno = ENAMETOOLONG;
+        return -1;
+    }
+    else {
+        return file_table[file].driver->mkdir(r, file_table[file].arg, path);
+    }
+}
+
+
+
+
+int _unlink(char *name) {
+    struct _reent* r = _REENT;
+    return file_table[file].driver->mkdir(r, file_table[file].arg, path);
+
+}
+
+int _link(char* old, char* new) {
+    
+    struct _reent* r = _REENT;
+    size_t i = 0;
+	for (i; i < MAX_FILELEN; i++) {
+		if (new[i] == '\0') {
+			break;
+		}
+	}
+
+    if(i >= MAX_FILELEN) {
```

Comment:
You have a couple places where you're checking the length of the file. Can you move them out into a function?

Reply:
Yah let me do that now.